File: 1.1/src/cz/mg/vulkan/PFNvkDestroyBufferView.java
package cz.mg.vulkan;

public class PFNvkDestroyBufferView extends VkFunctionPointer {
    public PFNvkDestroyBufferView() {
    }

    public PFNvkDestroyBufferView(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroyBufferView(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyBufferView(long value) {
        setValue(value);
    }

    public PFNvkDestroyBufferView(VkInstance instance) {
        super(instance, new VkString("vkDestroyBufferView"));
    }

    public void call(VkDevice device, VkBufferView bufferView, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), bufferView != null ? bufferView.getVkAddress() : VkPointer.getNullAddress(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long device, long bufferView, long pAllocator);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkDestroyBufferView.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyBufferView_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong bufferView, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyBufferView f = (PFN_vkDestroyBufferView)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkBufferView*)jniLongToPointer(bufferView)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

